### PR TITLE
Release tools - monorepo

### LINKED ISSUE #606: Adjust releasing tools to work with monorepo
Subissue of ckeditor/ckeditor5#6466.

Requires: https://github.com/ckeditor/ckeditor5-dev/issues/608.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Adjusted release tools to handle single mono-repository architecture. Closes #606.

### CODE REVIEW

**Comment by mlewand** [comment]
Also small UX thing: during the first use of `npm run release:bump-version -- --dry-run` I started to worry when the prompt was stuck for more than ~20s here:



Compared to previous packages build seemed to be stuck. After inspecting the changes I know now that it's because for builds we are are generating build package.

I was wondering whether we can quickly/cheaply append something like " (builds will take a longer time… relax)"?

**Comment by pomek** [comment]
> there's one issue that I found in release script though

https://github.com/ckeditor/ckeditor5/blob/master/scripts/release/update-utils-version.js#L12-L13

If we skip some steps of the release process, things must be done manually.

File: packages/ckeditor5-dev-env/lib/release-tools/utils/getpackagestorelease.js [hunk 1]
```diff
@@ -29,24 +32,43 @@ module.exports = function getPackagesToRelease( pathsToPackages ) {
...
 		}
 
 		return Promise.resolve();
 	}
 };
 
+// Search for a new version for specified package in the main changelog description.
+// It must be defined as following:
+//
+//      [packageName](https://www.npmjs.com/package/packageName): v0.0.1 => [v1.0.0](https://github.com/.../releases/tag/v1.0.0)
```
Comment (pomek) [inline packages/ckeditor5-dev-env/lib/release-tools/utils/getpackagestorelease.js:56]: TODO: Check whether it works with the following entries:

```md
* [@ckeditor/ckeditor5-word-count](https://www.npmjs.com/package/@ckeditor/ckeditor5-word-count): v0.0.2
* [@ckeditor/ckeditor5-build-balloon](https://www.npmjs.com/package/@ckeditor/ckeditor5-build-balloon): v19.0.0 => v19.0.1
```

Comment (pomek) [reply packages/ckeditor5-dev-env/lib/release-tools/utils/getpackagestorelease.js:56]: It won't Require changes. For upgrading existing repo:

```js
\[@ckeditor\/ckeditor5-word-count\].*: v[\d.]+ \=\> v([\d.]+)
```

For new release:

```js
\[@ckeditor\/ckeditor5-word-count\].*: v([\d.]+)
```

The order of execution is important. First must be checked an option with already existing package.

**Comment by pomek** [comment]
> I was wondering whether we can quickly/cheaply append something like " (builds will take a longer time… relax)"?

The tool does not know which scripts are being executed. We could add something like "echo Building..." to `preversion` hook in our builds.

File: packages/ckeditor5-dev-env/lib/release-tools/utils/getpackagestorelease.js (hunk 1 above)
Comment (pomek) [reply packages/ckeditor5-dev-env/lib/release-tools/utils/getpackagestorelease.js]: Done.

**Comment by mlewand** [comment]
> The tool does not know which scripts are being executed. We could add something like "echo Building..." to `preversion` hook in our builds.

Ok, so no cheap solution here. That being said, we can simply document this in our release instructions and this will be fine.

**Pull request merged**